bin/reth/src/node/mod.rs

```diff
@@ -126,7 +126,10 @@ impl Command {
 
         // Run pipeline
         info!(target: "reth::cli", "Starting sync pipeline");
-        pipeline.run(db.clone()).await?;
+        tokio::select! {
+            res = pipeline.run(db.clone()) => res?,
+            _ = tokio::signal::ctrl_c() => {},
```

Comment:
this is where we need to handle things like flush to disk, so here we need to `GetPeers` write to file.

Reply:
ACK - this was my bad then @MegaRedHand

Reply:
😄 No problem! I also didn't know about `drop` not triggering 🤯 
Just finished addressing all comments, and moved the flush back here.